### PR TITLE
hasParent Query Parameter for GET /api/document

### LINKED ISSUE #21: Add ability to get all root documents
Add ability to get all documents that have no parent.

### PR DESCRIPTION
**Pull request by alongreyber**
Added ability to use the hasParent query parameter on the /api/document endpoint. Also updated tests and API documentation. Fixes #21.

### CODE REVIEW

File: api/openapi.yml [hunk 1]
```diff
@@ -15,6 +15,12 @@ paths:
     get:
       summary: Get documents
       description: Get a list of documents that are either public or associated with the current user
+      parameters:
+        - in: query
+          name: hasParent
```
Comment (patrickkwang) [inline api/openapi.yml:20]:
```suggestion
          name: has_parent
```
https://api.gov.au/standards/national_api_standards/naming-conventions.html#message-format

File: internal/robokache/server.go [hunk 2]
```diff
@@ -31,6 +32,11 @@ func AddGUI(r *gin.Engine) {
 	r.Static("/docs", "./api")
 }
 
+// Query parameters for Document get request
+type GetDocumentQuery struct {
+	HasParent *bool `form:"hasParent"`
```
Comment (patrickkwang) [inline internal/robokache/server.go:37]:
```suggestion
	HasParent *bool `form:"has_parent"`
```

**Review by patrickkwang — CHANGES_REQUESTED** [review]

File: api/openapi.yml (hunk 1 above)
Comment (alongreyber) [reply api/openapi.yml]: Fixed

File: internal/robokache/server.go (hunk 2 above)
Comment (alongreyber) [reply internal/robokache/server.go]: Fixed

**Pull request merged**